src/content/doc-surrealql/datamodel/idioms.mdx

```diff
@@ -0,0 +1,601 @@
+---
+sidebar_position: 10
+sidebar_label: Idioms
+title: Idioms | SurrealQL
+description: Accessing and manipulating data using idioms (paths) in SurrealQL.
+---
+
+import Since from '@components/shared/Since.astro'
+
+# Idioms
+
+Idioms in SurrealQL provide a powerful and flexible way to access and manipulate data within records using paths. They allow you to navigate through nested data structures, access fields, array elements, call methods, and perform complex queries with ease. Idioms are similar to expressions in other query languages that provide a path to data within documents or records.
+
+An idiom is composed of a sequence of **parts** that define the path to a value within a record or data structure. Each part specifies how to navigate to the next piece of data. Idioms can be used in various parts of SurrealQL. The most common usecase is in data retrival queries such as `SELECT` statements, but they can also be used in the `WHERE` clause, `SET` clause, and more.
+
+## Parts of an Idiom
+
+An idiom is made up of one or more **parts**, each of which can be one of several types:
+
+- [**Field**](#field-access): Access a field by name.
+- [**Index**](#index-access): Access an element of an array by its index.
+- [**All**](#all-elements): Access all elements or fields.
+- [**Last**](#last-element): Access the last element of an array.
+- **First**: Access the first element of an array.
```

Comment:
This is the only line without a link.

Reply:
Removing the first section as it is implicit behaviour